### ISSUE TITLE
[Feature]: Add deep infra as a provider 

### ISSUE THREAD

**Issue by ishaan-jaff** [BerriAI/litellm#450]
### The Feature

cc @shauryr

### Motivation, pitch

support for DeepInfra. This is the easiest and cheapest way to get llama 2 running for your system. They support openai api right now - https://deepinfra.com/meta-llama/Llama-2-70b-chat-hf/api?example=openai-python
I imagine calling it via litellm would be bet

**Comment by ranjancse26** [comment]
Alternatively, one can do like this.

```
import litellm
os.environ['OPENAI_API_KEY'] = "<<API_KEY>>"
messages = [{"role":"user","content":"Request boss to grant me 1 day leave"}]
litellm.api_base = "https://api.deepinfra.com/v1/"
response = litellm.completion(messages=messages, model="gpt-3.5-turbo")

**Comment by ishaan-jaff** [comment]
pushed a commit

**Comment by ishaan-jaff** [comment]
usage 

```python
import os
from litellm import completion

os.environ['DEEPINFRA_API_KEY'] = "" 
model_name = "deepinfra/meta-llama/Llama-2-70b-chat-hf"
try:
    response = completion(model=model_name, messages=messages)
    # Add any assertions here to check the response
    print(response)
```

**Comment by ishaan-jaff** [comment]
litellm 0.1.758 has support for deep infra 

pending work 

- [x] add streaming
- [x] add exception mapping 
- [x] add docs

**Comment by ishaan-jaff** [comment]
Deep infra already maps the exceptions to openai exceptions, 

Tried this for bad API key errors

**Comment by ishaan-jaff** [comment]
re streaming: they support streaming but they have one bug on their server:

their streaming responses have role after first chunk

I notified their team to fix this

**Comment by ghost** [comment]
@ishaan-jaff looks like this is done?https://docs.litellm.ai/docs/providers/deepinfra#sample-usage---streaming

**Comment by ishaan-jaff** [comment]
streaming is done: pushed to prod too 

(they have one bug with they way their sending chunks) but it's good for using @shauryr

**Comment by ishaan-jaff** [comment]
usage 
```python
import litellm
import os

os.environ['DEEPINFRA_API_KEY'] = ""
response = completion(
    model="deepinfra/meta-llama/Llama-2-70b-chat-hf", 
    messages=[{"role": "user", "content": "write code for saying hi from LiteLLM"}],
    stream=True
)

for chunk in response:
    print(chunk)
```

**Comment by shauryr** [comment]
Maybe I am doing something wrong here. I am on litellm==0.1.7713
Please let me know how can I fix this - 

<img width="692" alt="image" src="https://github.com/BerriAI/litellm/assets/12604876/e7599473-7e81-4bc3-8a7f-7cc26b971bfc">

**Comment by shauryr** [comment]
Maybe I am doing something wrong here. Please let me know - 

<img width="1074" alt="image" src="https://github.com/BerriAI/litellm/assets/12604876/86a8fddd-f7d5-4310-b579-507622e3e987">

**Comment by shauryr** [comment]
I am on version: litellm==0.1.7713

**Comment by ishaan-jaff** [comment]
@shauryr what LiteLLM version are you on? 
Can you try bumping to the latest litellm 0.1.7713


Can you verify this code snippet works for you? 
```python
import litellm
import os

os.environ['DEEPINFRA_API_KEY'] = ""
response = completion(
    model="deepinfra/meta-llama/Llama-2-70b-chat-hf", 
    messages=[{"role": "user", "content": "write code for saying hi from LiteLLM"}],
    stream=True
)

for chunk in response:
    print(chunk)
```

**Comment by ishaan-jaff** [comment]
I can verify this works for me on litellm:

```python
import litellm

import os

os.environ['DEEPINFRA_API_KEY'] = ""
response = litellm.completion(
    model="deepinfra/meta-llama/Llama-2-70b-chat-hf", 
    messages=[{"role": "user", "content": "write code for saying hi from LiteLLM"}],
    stream=True
)

for chunk in response:
    print(chunk)

```

**Comment by shauryr** [comment]
yup works for me -

<img width="1056" alt="image" src="https://github.com/BerriAI/litellm/assets/12604876/73691416-60d4-4a8e-bd05-aa57b7618fde">

**Comment by ishaan-jaff** [comment]
so looks like this is a bug with the Llama index integration , i can repro the error on my side too now

**Comment by shauryr** [comment]
ahh! Thanks for looking into this though. I appreciate the promptness 🙂

**Comment by ishaan-jaff** [comment]
ahh i see the error, it's caused by this: https://github.com/jerryjliu/llama_index/blob/415e9c0978bae0f74c1d1d31cdd26f68444c90fd/llama_index/llms/litellm_utils.py#L189 

cc @krrishdholakia

**Comment by ghost** [comment]
@shauryr fix pushed here - 076affbff28c7094c0a63cf0a2110a1b7b8ad622

will take 10 minutes to be in prod (running through testing)

**Comment by ghost** [comment]
in prod - `pip install litellm==0.1.774` @shauryr
